src/main/scala/outwatch/dom/helpers/DomUtils.scala

```diff
@@ -144,48 +136,48 @@ private[outwatch] final case class SeparatedEmitters(
   def ::(e: Emitter): SeparatedEmitters = copy(emitters = e :: emitters)
 }
 
+
+private[outwatch] case class VNodeState(
+  nodes: Array[Seq[IO[StaticVNode]]],
+  attributes: Map[String, Attribute] = Map.empty
+)
+
+private[outwatch] object VNodeState {
+  def from(nodes: List[ChildVNode]): VNodeState = VNodeState(
+    nodes.map {
+      case n: StaticVNode => List(IO.pure(n))
+      case _ => List.empty
+    }(breakOut)
+  )
+
+  type Updater = VNodeState => VNodeState
+}
+
 private[outwatch] final case class Receivers(
   children: Children,
   attributeStreamReceivers: List[AttributeStreamReceiver]
 ) {
-  private val (childNodes, childStreamStatus) = children match {
-    case Children.VNodes(nodes, streamStatus) => (nodes, streamStatus)
-    case _ => (Nil, Children.StreamStatus())
+  private val (nodes, hasNodeStreams) = children match {
+    case Children.VNodes(nodes, hasStream) => (nodes, hasStream)
+    case _ => (Nil, false)
   }
 
-  lazy val observable: Observable[(Seq[Attribute], Seq[IO[StaticVNode]])] = {
-    val childStreamReceivers = if (childStreamStatus.hasChildOrChildren) {
-      childNodes.foldRight(Observable(List.empty[IO[StaticVNode]])) {
-        case (vn: StaticVNode, obs) => obs.combineLatestMap(BehaviorSubject(IO.pure(vn)))((nodes, n) => n :: nodes)
-        case (csr: ChildStreamReceiver, obs) =>
-          obs.combineLatestMap(
-            if (childStreamStatus.hasMultipleChildOrChildren) csr.childStream.startWith(Seq(IO.pure(StringVNode("")))) else csr.childStream
-          )((nodes, n) => n :: nodes)
-        case (csr: ChildrenStreamReceiver, obs) =>
-          obs.combineLatestMap(
-            if (childStreamStatus.hasMultipleChildOrChildren) csr.childrenStream.startWith(Seq(Seq.empty)) else csr.childrenStream
-          )((nodes, n) => n.toList ++ nodes)
-      }
-    } else {
-      Observable(Seq.empty)
-    }
-
-    // only use last encountered observable per attribute
```

Comment:
Do we have a test for this?

Reply:
This is very much at the core of the library, all tests that contain VDOM Observables are tests for it.

Reply:
I was referring to:
```
// only use last encountered observable per attribute
```
I was just wondering if we have a test for this, else I would write one.

Reply:
Yes, it's tested in a couple of places:

https://github.com/OutWatch/outwatch/blob/master/src/test/scala/outwatch/OutWatchDomSpec.scala#L744

https://github.com/OutWatch/outwatch/blob/master/src/test/scala/outwatch/OutWatchDomSpec.scala#L725